### PR TITLE
feat: improve default dh.ui layouts

### LINKED ISSUE #317: ui.flex defaults

### PR DESCRIPTION
**Pull request by dsmmcken**
- ui.panel now has default padding and gaps
- ui.panel now accepts a subset of flex and view props
- ui.panel now wraps children in view and flex containers
- ui.panels with only one child (of type grid or plot) have padding removed
- ui.flex flex property now defaults to "auto"
- ui.flex/ui.view switched to use the wrapped @deephaven/component version
- updated package.json to use latest linting rules (to ignore UNSAFE casing)
- Fixes #317 

Breaking changes:
- ui.panel now has default padding and gaps
- ui.flex flex property now defaults to "auto"

### CODE REVIEW

File: plugins/ui/src/js/src/elements/spectrum/View.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
Comment (bmingles) [inline plugins/ui/src/js/src/elements/spectrum/View.tsx:1]: We should be able to delete the `Flex` and `View` wrappers and just `import { Flex, View } from '@deephaven/components'` directly in `SpectrumElementUtils`

File: plugins/ui/src/deephaven/ui/components/spectrum/flex.py [hunk 2]
```diff
@@ -24,12 +26,24 @@ def flex(
...
-    https://react-spectrum.adobe.com/react-spectrum/Flex.html
+    Base Flex component for laying out children in a flexbox.
+
+    Args:
+        children: Elements to render in the flexbox.
+        flex: The flex property of the flexbox.
+        direction: The direction in which to layout children.
+        wrap: Whether children should wrap when they exceed the panel's width.
+        justify_content: The distribution of space around items along the main axis.
+        align_content: The distribution of space between and around items along the cross axis.
```
Comment (bmingles) [inline plugins/ui/src/deephaven/ui/components/spectrum/flex.py:37]: Is the mismatch of arg comments with explicit args intentional?

Comment (dsmmcken) [reply plugins/ui/src/deephaven/ui/components/spectrum/flex.py:37]: what do you mean?

**Review by bmingles — APPROVED** [review]
LGTM

File: plugins/ui/src/js/src/layout/ReactPanel.tsx [hunk 3]
```diff
@@ -117,7 +169,38 @@ function ReactPanel({ children, title }: ReactPanelProps): JSX.Element | null {
...
+            paddingStart={paddingStart}
+            paddingEnd={paddingEnd}
+            paddingX={paddingX}
+            paddingY={paddingY}
+            UNSAFE_style={UNSAFE_style}
+            UNSAFE_className={
+              UNSAFE_className == null
+                ? 'dh-react-panel'
+                : `${UNSAFE_className} dh-react-panel`
+            }
```
Comment (mofojed) [inline plugins/ui/src/js/src/layout/ReactPanel.tsx:187]: I guess we don't have the `classnames` package here eh. I wouldn't be opposed to adding it, though unnecessary for this.
Alternative:
```
UNSAFE_className={`${UNSAFE_className ?? ''} dh-react-panel`}
```

**Review by mofojed — CHANGES_REQUESTED** [review]
Pull the `playwright` change into a separate PR

File: plugins/ui/src/js/src/layout/ReactPanel.tsx (hunk 3 above)
Comment (dsmmcken) [reply plugins/ui/src/js/src/layout/ReactPanel.tsx:187]: Not that it matters, but that leaves you with a leading space.

File: package.json [hunk 4]
```diff
@@ -26,7 +26,7 @@
     "@deephaven/eslint-config": "^0.72.0",
     "@deephaven/prettier-config": "^0.72.0",
     "@deephaven/tsconfig": "^0.72.0",
-    "@playwright/test": "^1.41.2",
+    "@playwright/test": "^1.43.1",
```
[the remark this answers is not in the document]
Comment (dsmmcken) [reply package.json:29]: reverted, I was flailing at things to get playwright to pass.

**Review by mofojed — APPROVED** [review]

File: plugins/ui/src/js/src/layout/ReactPanel.tsx (hunk 3 above)
Comment (mofojed) [reply plugins/ui/src/js/src/layout/ReactPanel.tsx:187]: It does, but also leave you without duplicating the text `dh-react-panel` twice without any sort of type guard.

**Pull request merged**